src/dataset-builder/ConceptSetCreator.ts

```diff
@@ -32,12 +41,17 @@ export const ConceptSetCreator = (props: ConceptSetCreatorProps) => {
     children: _.map(_.get('root'), snapshotBuilderSettings?.domainOptions),
   };
   return h(ConceptSelector, {
-    rootConcept: domainOptionRoot,
-    initialCart: [],
+    // Concept selection currently only supports top level domains, so nodes should not be expandable
+    rootConcept: {
+      ...domainOptionRoot,
+      children: _.map((child) => ({ ...child, hasChildren: false }), domainOptionRoot.children),
+    },
+    initialCart: cart,
     title: 'Add concept',
     onCancel: () => onStateChange(homepageState.new()),
     onCommit: (selected: Concept[]) => {
-      conceptSetUpdater((conceptSets) => _.flow(_.map(toConceptSet), _.union(conceptSets))(selected));
+      // ensures cart persistence without duplicating items in the cart
+      conceptSetUpdater(() => _.flow(_.map(toConceptSet))(selected));
```

Comment:
I don't think `flow` is doing much for us here. Before it was composing the result of two operations (`union()` and `map()`) but now it's only calling `map()`, so it would be clearer to call `map()` directly.
```suggestion
      conceptSetUpdater(() => _.map(toConceptSet, selected));
```